AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -387,9 +392,21 @@ protected Void doInBackground(Void... params) {
   }
 
   private void sendRequests() {
-    List<Map<String, Object>> unsentRequests = getUnsentRequests();
-    List<Map<String, Object>> requestsToSend =
-        removeIrrelevantBackgroundStartRequests(unsentRequests);
+    List<Map<String, Object>> unsentRequests = new ArrayList<>();
+    List<Map<String, Object>> requestsToSend;
+    // Check if we have localErrors, if yes then we will send only errors to the server.
+    if (localErrors.size() != 0) {
```

Comment:
I wonder if we can abstract this a little bit and reuse it here for the error logic and for the regular events?